client.go

```diff
@@ -0,0 +1,131 @@
+	}
+	endpoint := fmt.Sprintf("%s:%d", cfg.IP, cfg.Port)
+	listener, err := net.NewTCPListener(endpoint)
+	if err != nil {
+		return nil, errors.WithMessagef(err, "listening on %s", endpoint)
+	}
+	dialer := net.NewTCPDialer(time.Second * 15)
+	node, err := ethclient.Dial(cfg.ETHNodeURL)
+	if err != nil {
+		return nil, errors.WithMessage(err, "connecting to ethereum node")
+	}
+	cb := ethchannel.NewContractBackend(node, w.Ks, &acc.Account)
+	adjudicator := ethchannel.NewAdjudicator(cb, adjudicatorAdr, acc.Account.Address)
+	funder := ethchannel.NewETHFunder(cb, assetAddr)
+
+	client := client.New(acc, dialer, funder, adjudicator, w)
+	go client.Listen(listener)
+	return &Client{cfg: cfg, client: client, w: w, onChain: acc, dialer: dialer}, nil
+}
+
+func unlockAccount(cfg *Config) ethwallet.Account {
+	return ethwallet.Account{}
+}
+
+func importAccount(walletPath, secret string) (*ethwallet.Wallet, *ethwallet.Account, error) {
+	ks := keystore.NewKeyStore(walletPath, 2, 1)
+	sk, err := crypto.HexToECDSA(secret[2:])
+	if err != nil {
+		return nil, nil, errors.WithMessage(err, "decoding secret key")
+	}
+	var ethAcc accounts.Account
+	addr := crypto.PubkeyToAddress(sk.PublicKey)
+	if ethAcc, err = ks.Find(accounts.Account{Address: addr}); err != nil {
+		ethAcc, err = ks.ImportECDSA(sk, "")
+		if err != nil && errors.Cause(err).Error() != "account already exists" {
+			return nil, nil, errors.WithMessage(err, "importing secret key")
+		}
+	}
+
+	w, err := ethwallet.NewWallet(ks, "")
+	if err != nil {
+		return nil, nil, errors.WithMessage(err, "creating wallet")
+	}
+
+	wAcc := ethwallet.NewAccountFromEth(w, &ethAcc)
+	acc, err := w.Unlock(wAcc.Address())
+	return w, acc.(*ethwallet.Account), err
+}
+
+func (c *Client) AddPeer(perunID *Address, host string, port int) {
+	c.dialer.Register((*ethwallet.Address)(&perunID.addr), fmt.Sprintf("%s:%d", host, port))
+}
+
+func (c *Client) ProposeChannel(ctx *Context, perunID *Address, challengeDuration int64, initialBals *BigInts) (*PaymentChannel, error) {
+	if initialBals.Length() != 2 {
+		return nil, errors.New("initBals needs exactly two entries")
+	}
+	alloc := &channel.Allocation{
+		Assets:   []channel.Asset{(*ethwallet.Address)(&assetAddr)},
+		Balances: [][]channel.Bal{{initialBals.values[0], initialBals.values[1]}},
+	}
+	prop := &client.ChannelProposal{
+		ChallengeDuration: uint64(challengeDuration),
+		Nonce:             nonce(),
+		ParticipantAddr:   c.w.NewAccount().Address(),
+		AppDef:            payment.AppDef(),
+		InitData:          &payment.NoData{},
+		InitBals:          alloc,
+		PeerAddrs:         []wallet.Address{c.onChain.Address(), (*ethwallet.Address)(&perunID.addr)},
+	}
+	_ch, err := c.client.ProposeChannel(ctx.ctx, prop)
+	return &PaymentChannel{_ch}, err
+}
+
+// nonce generates a cryptographically secure random value in the range [0, 2^256 -1]
+func nonce() *big.Int {
+	max := new(big.Int)
+	max.Lsh(big.NewInt(1), 256).Sub(max, big.NewInt(1))
```

Comment:
You can actually set a global `var max *big.Int` variable once in a `init()` function that you create in this file. Then better name it `maxUint256`.